src/commands/registry.rs

```diff
@@ -57,6 +57,10 @@ pub struct Push {
     /// This is a string whose format is defined by the registry standard, and generally consists of <registry>/<username>/<application-name>:<version>. E.g. ghcr.io/ogghead/spin-test-app:0.1.0
     #[clap()]
     pub reference: String,
+
+    /// Cache directory for downloaded registry data.
+    #[clap(long)]
+    pub download_cache_dir: Option<PathBuf>,
```

Comment:
Can we just call it `cache_dir` or do we expect have more than one on-disk cache?

Reply:
There is already a `--cache` option for the Wasmtime cache configuration file.  I wanted to distinguish this cache from that.